src/reducers/gifsFavorites.js

```diff
@@ -0,0 +1,48 @@
+import * as types from '../constants/actionTypes';
+import _localStorage from '../localStorage';
+
+const getLocalStorage = () => {
+  if(window.localStorage) return localStorage;
+  return _localStorage;
+}
+
+const _favs = JSON.parse(getLocalStorage().getItem('favorites') || '[]');
+const initialState = {
+  favorites: _favs,
+  textFilter: '',
+};
+
+const gifsFavorites = (state = initialState, action) => {
+  const { type, gif } = action;
+  switch (type) {
+    case types.GIF_FAVORITE_ADD: {
+      const newState = { ...state };
+      const favorites = [...newState.favorites];
+      if (!favorites.find(x => x.id == gif.id)) favorites.push(gif);
+      newState.favorites = favorites;
+      getLocalStorage().setItem('favorites', JSON.stringify(newState.favorites));
```

Comment:
Here you are calling `localStorage`, this is a side effect, something that we should avoid in the reducers. You can achieve this feature using middleware, sagas or a subscription to the store.